### ISSUE TITLE
restore operator crashed when restoring

### ISSUE THREAD

**Issue by xcompass** [on2itsecurity/etcd-operator#80]
Any ideas? Tried both 1.2.0 and 1.2.1. (Though it still show Version 1.2.0 when using 1.2.1)

```
time="2022-06-27T18:04:48Z" level=info msg="Go Version: go1.18.2"
time="2022-06-27T18:04:48Z" level=info msg="Go OS/Arch: linux/amd64"
time="2022-06-27T18:04:48Z" level=info msg="etcd-restore-operator Version: 1.2.0"
time="2022-06-27T18:04:48Z" level=info msg="Git SHA: ef26a688b09be4e9445b33bad5d76da8c6020daf"
I0627 18:04:48.310663       1 leaderelection.go:248] attempting to acquire leader lease default/etcd-restore-operator...
I0627 18:04:48.318244       1 leaderelection.go:258] successfully acquired lease default/etcd-restore-operator
time="2022-06-27T18:04:48Z" level=info msg="listening on 0.0.0.0:19999"
time="2022-06-27T18:04:48Z" level=info msg="starting restore controller" pkg=controller

E0627 18:19:15.551844       1 runtime.go:79] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 68 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x1bb2dc0?, 0x3153c40})
        /go/src/github.com/on2itsecurity/etcd-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:75 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc00066b1c0?})
        /go/src/github.com/on2itsecurity/etcd-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:49 +0x75
panic({0x1bb2dc0, 0x3153c40})
        /usr/local/go/src/runtime/panic.go:838 +0x207
github.com/on2itsecurity/etcd-operator/pkg/util/k8sutil.NewSeedMemberPod({0x229ba40, 0xc000130008}, {0x22aa120, 0xc000904c00}, {0xc000802ce0, 0xf}, {0xc000802cb6, 0x7}, _, 0xc000450e00, ...)
        /go/src/github.com/on2itsecurity/etcd-operator/pkg/util/k8sutil/k8sutil.go:318 +0x174
github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator.(*Restore).createSeedMember(0xc00096f560, {0x229ba40, 0xc000130008}, 0xc0009b4b40, {0xc00091a9f0, 0x23}, {0xc000802cb6, 0x7}, {0xc000802ce0, 0xf}, ...)
        /go/src/github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator/sync.go:227 +0x687
github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator.(*Restore).prepareSeed(0xc00096f560, {0x229ba40, 0xc000130008}, 0xc0000e2180)
        /go/src/github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator/sync.go:187 +0x9ae
github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator.(*Restore).handleCR(0xc00096f560, {0x229ba40, 0xc000130008}, 0xc0000e2180, {0xb0cb8a?, 0xc0006cd0b0?})
        /go/src/github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator/sync.go:90 +0x20f
github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator.(*Restore).processItem(0xc00096f560, {0x229ba40, 0xc000130008}, {0xc0009c0be8, 0x17})
        /go/src/github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator/sync.go:70 +0x7e
github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator.(*Restore).processNextItem(0xc00096f560, {0x229ba40, 0xc000130008})
        /go/src/github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator/sync.go:56 +0xe8
github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator.(*Restore).runWorker(0xc0005096a0?)
        /go/src/github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator/sync.go:42 +0x33
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x0?)
        /go/src/github.com/on2itsecurity/etcd-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x0?, {0x22848c0, 0xc0009b9290}, 0x1, 0xc00010f020)
        /go/src/github.com/on2itsecurity/etcd-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0005097b0?, 0x3b9aca00, 0x0, 0x0?, 0xb72b60?)
        /go/src/github.com/on2itsecurity/etcd-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(0x8b47aa?, 0xa0096a101?, 0xc0005097a0?)
        /go/src/github.com/on2itsecurity/etcd-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x25
created by github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator.(*Restore).run
        /go/src/github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator/controller.go:55 +0x70f
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xb0 pc=0x1978df4]

goroutine 68 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc00066b1c0?})
        /go/src/github.com/on2itsecurity/etcd-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:56 +0xd8
panic({0x1bb2dc0, 0x3153c40})
        /usr/local/go/src/runtime/panic.go:838 +0x207
github.com/on2itsecurity/etcd-operator/pkg/util/k8sutil.NewSeedMemberPod({0x229ba40, 0xc000130008}, {0x22aa120, 0xc000904c00}, {0xc000802ce0, 0xf}, {0xc000802cb6, 0x7}, _, 0xc000450e00, ...)
        /go/src/github.com/on2itsecurity/etcd-operator/pkg/util/k8sutil/k8sutil.go:318 +0x174
github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator.(*Restore).createSeedMember(0xc00096f560, {0x229ba40, 0xc000130008}, 0xc0009b4b40, {0xc00091a9f0, 0x23}, {0xc000802cb6, 0x7}, {0xc000802ce0, 0xf}, ...)
        /go/src/github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator/sync.go:227 +0x687
github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator.(*Restore).prepareSeed(0xc00096f560, {0x229ba40, 0xc000130008}, 0xc0000e2180)
        /go/src/github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator/sync.go:187 +0x9ae
github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator.(*Restore).handleCR(0xc00096f560, {0x229ba40, 0xc000130008}, 0xc0000e2180, {0xb0cb8a?, 0xc0006cd0b0?})
        /go/src/github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator/sync.go:90 +0x20f
github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator.(*Restore).processItem(0xc00096f560, {0x229ba40, 0xc000130008}, {0xc0009c0be8, 0x17})
        /go/src/github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator/sync.go:70 +0x7e
github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator.(*Restore).processNextItem(0xc00096f560, {0x229ba40, 0xc000130008})
        /go/src/github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator/sync.go:56 +0xe8
github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator.(*Restore).runWorker(0xc0005096a0?)
        /go/src/github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator/sync.go:42 +0x33
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x0?)
        /go/src/github.com/on2itsecurity/etcd-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x0?, {0x22848c0, 0xc0009b9290}, 0x1, 0xc00010f020)
        /go/src/github.com/on2itsecurity/etcd-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0005097b0?, 0x3b9aca00, 0x0, 0x0?, 0xb72b60?)
        /go/src/github.com/on2itsecurity/etcd-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(0x8b47aa?, 0xa0096a101?, 0xc0005097a0?)
        /go/src/github.com/on2itsecurity/etcd-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x25
created by github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator.(*Restore).run
        /go/src/github.com/on2itsecurity/etcd-operator/pkg/controller/restore-operator/controller.go:55 +0x70f
```

Here is the restore CR
```
apiVersion: "etcd.database.coreos.com/v1beta2"
kind: "EtcdRestore"
metadata:
  # The restore CR name must be the same as spec.etcdCluster.name
  name: gp-etcd-cluster
spec:
  etcdCluster:
    # The namespace is the same as this EtcdRestore CR
    name: gp-etcd-cluster
  backupStorageType: S3
  s3:
    # The format of "path" must be: "<s3-bucket-name>/<path-to-backup-file>"
    # e.g: "mybucket/etcd.backup"
    path: "xxxxx/default/gp-etcd-cluster-hourly_v3583558_2022-06-26-09:53:17"
    awsSecret: gp-etcd-aws
    endpoint: ""
    forcePathStyle: false
````

**Comment by nberlee** [comment]
Could you please test the container image which is referenced here: https://github.com/on2itsecurity/etcd-operator/pkgs/container/etcd-operator/26681817?tag=216a9eed4fdd87bf8ecd66f8548dff3260af4ee3

You only need to replace the restore-operators image.
Do not get you hopes up yet :) this should only prevent the panic from happening. At least at this point in the code. The real error message should be visible now in the code.

**Comment by nberlee** [comment]
image: `ghcr.io/on2itsecurity/etcd-operator:b9f32f3c27987119d36ab03743866a083ebe1f2a`
should at give further info

**Comment by xcompass** [comment]
Thanks for the quick fix. It works for me! I was able to restore a backup from S3 with the new image without any error.

**Comment by nberlee** [comment]
Fix is part of 1.2.2 which is just releaed
